apps/sensenet/src/components/appbar/desktop-nav-menu.tsx

```diff
@@ -65,44 +92,66 @@ export const DesktopNavMenu: React.FunctionComponent = () => {
   const globalClasses = useGlobalStyles()
   const theme = useTheme()
   const service = injector.getInstance(PersonalSettings)
-  const { openDialog } = useDialog()
-  const [open, setOpen] = useState(false)
-  const anchorRef = useRef<HTMLButtonElement>(null)
   const currentUser = useCurrentUser()
   const repo = useRepository()
   const localization = useLocalization()
+  const { openDialog } = useDialog()
+  const [openLogoutMenu, setOpenLogoutMenu] = useState(false)
+  const [openHelpMenu, setOpenHelpMenu] = useState(false)
+  const logoutRef = useRef<HTMLButtonElement>(null)
+  const helpRef = useRef<HTMLButtonElement>(null)
 
-  const handleToggle = () => {
-    setOpen((prevOpen) => !prevOpen)
+  const handleToggle = (setter: React.Dispatch<React.SetStateAction<boolean>>) => {
+    setter((prevState) => !prevState)
   }
 
-  const handleClose = (event: React.MouseEvent<EventTarget>) => {
-    if (anchorRef.current && anchorRef.current.contains(event.target as HTMLElement)) {
+  const handleClose = (
+    event: React.MouseEvent<EventTarget>,
+    ref: React.RefObject<HTMLButtonElement>,
+    setter: React.Dispatch<React.SetStateAction<boolean>>,
+  ) => {
+    if (ref.current && ref.current.contains(event.target as HTMLElement)) {
       return
     }
 
-    setOpen(false)
+    setter(false)
   }
 
   function handleListKeyDown(event: React.KeyboardEvent) {
     if (event.key === 'Tab') {
       event.preventDefault()
-      setOpen(false)
+      setOpenLogoutMenu(false)
+      setOpenHelpMenu(false)
     }
   }
 
-  const prevOpen = useRef(open)
+  const prevLogoutOpen = useRef(openLogoutMenu)
+  const prevHelpOpen = useRef(openHelpMenu)
+
+  useEffect(() => {
```

Comment:
why is this magic useful?

Reply:
https://material-ui.com/components/popper/#SimplePopper.tsx